packages/artifact/docs/implementation-details.md

```diff
@@ -0,0 +1,43 @@
+# Implementation Details
```

Comment:
Don't have to be in this PR, but I think we should call out explicitly that this is not a contract we are going to maintain - it's great to be transparent about the internal choices we make, but at the same time we are free to make any changes here.  Don't code any solution based on information in this doc.  For example, if the customer scenario warrants compressing the artifacts, please do it themselves, we may not compress for them in the future.

Reply:
I'll add information about this in my next PR when I'm adding contribution guidelines and tests